src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -0,0 +1,276 @@
+                                        s => s.Name == "OnAuthorizationAsync" &&
+                                            s.ReturnType.Equals(taskTypeSymbol) &&
+                                            s.Parameters.Length == 1 &&
+                                            s.Parameters[0].Type.Equals(authorizationFilterContextTypeSymbol)));
+                            }
+                        }
+                    }
+                }, OperationKind.Invocation);
+
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolAnalysisContext =>
+                {
+                    var controllerTypeSymbol = (INamedTypeSymbol)symbolAnalysisContext.Symbol;
+
+                    // The controller class is protected by a validate anti forgery token attribute
+                    if (controllerTypeSymbol.GetAttributes().Any(s => s_AntiForgeryAttributeRegex.IsMatch(s.AttributeClass.Name)))
+                    {
+                        usingValidateAntiForgeryAttribute = true;
+
+                        return;
+                    }
+
+                    foreach (var actionMethodSymbol in controllerTypeSymbol.GetMembers().OfType<IMethodSymbol>())
+                    {
+
+                        // The method is protected by a validate anti forgery token attribute
+                        if (actionMethodSymbol.GetAttributes().Any(s => s_AntiForgeryAttributeRegex.IsMatch(s.AttributeClass.Name)))
+                        {
+                            usingValidateAntiForgeryAttribute = true;
+
+                            return;
+                        }
+
+                        var baseTypes = controllerTypeSymbol.GetBaseTypes();
+
+                        if ((baseTypes.Contains(controllerTypeSymbol) ||
+                            baseTypes.Contains(controllerBaseTypeSymbol)) && // An subtype of `Microsoft.AspNetCore.Mvc.Controller` or `Microsoft.AspNetCore.Mvc.ControllerBase`
+                            actionMethodSymbol.IsPublic() &&
+                            !actionMethodSymbol.IsStatic &&
+                            !actionMethodSymbol.HasAttribute(nonActionAttributeTypeSymbol))
+                        {
+                            if (actionMethodSymbol.HasAttribute(httpPostAttributeTypeSymbol))
+                            {
+                                actionMethodSymbols.Add((actionMethodSymbol, "HttpPost"));
+                            }
+                            else if (actionMethodSymbol.HasAttribute(httpPutAttributeTypeSymbol))
+                            {
+                                actionMethodSymbols.Add((actionMethodSymbol, "HttpPut"));
+                            }
+                            else if (actionMethodSymbol.HasAttribute(httpDeleteAttributeTypeSymbol))
+                            {
+                                actionMethodSymbols.Add((actionMethodSymbol, "HttpDelete"));
+                            }
+                        }
```

Comment:
I think we should handle HttpPatchAttribute as well, since that's a write operation.